docs/base/dump_plugin.rst

```diff
@@ -0,0 +1,43 @@
+.. _craft_parts_dump_plugin:
+
+Dump plugin
+=============
+
+The Dump plugin can be used for any project where you want to include existing
+files from somewhere and keep the content as is. Its source can be a local
+directory, a remote repository, or a URL. Common use cases include:
+
+- Include static files like scripts or media from a local directory.
+- Download and unpack a pre-compiled proprietary software from a remote URL.
+- Use git to clone a remote SDK, dataset, or model.
+
+
+Keywords
+--------
+
+This plugin uses the common :ref:`plugin <part-properties-plugin>` keywords as
+well as those for :ref:`sources <part-properties-sources>`.
+
+You will need to specify at least the :ref:`source <source>` keyword.
+:ref:`source-type <source_type>` keyword is optional but recommended for
+specific how to handle the source.
+
+
+Dependencies
+------------
+
+This plugin has no dependencies.
+
+
+How it works
+------------
+
+During the build step, the plugin performs the following actions:
```

Comment:
I think this list is misleading: as far as I know every bullet point [except the last](https://github.com/canonical/craft-parts/blob/main/craft_parts/plugins/dump_plugin.py#L82) is actually handled by the source type, and not the plugin. That is, almost everything here is shared by all plugins, including the `nil` plugin

Reply:
They are shared, but that doesn't affect how to explain to the user what the plugin does. You can't say that this plugin copies files and then doesn't explain to the user how those files got there.

Reply:
I'm fine leaving it as-is, but perhaps briefer version would be:

```
During the build step, the plugin copies the entire directory tree from the source to the part's install directory.
```

It is probably out of scope to explain the source type here, and we reference the [part source](https://canonical-craft-parts.readthedocs-hosted.com/en/latest/explanation/parts.html#source) to send more details of how the source directory gets filled.